### PR TITLE
SUSEConnect over proxy server with authentication

### PR DESCRIPTION
**Pull request by jenting**
Signed-off-by: JenTing Hsiao <jenting.hsiao@suse.com>

### CODE REVIEW

File: adoc/deployment-preparation.adoc [hunk 1]
```diff
@@ -89,6 +89,16 @@ sudo zypper in -t pattern SUSE-CaaSP-Management
 This will install the `skuba` command line tool and `{tf}`; as well
 as various default configurations and examples.
 
+[NOTE] Using a Proxy Server with Authentication
+====
+Sometimes, you need a proxy server to be able to connects the SUSE Customer Center and the proxy server requires authentication. Create the file `/root/.curlrc` with the content:
+----
+--proxy https://PROXY_FQDN:PROXY_PORT
```
Comment (innobead) [inline adoc/deployment-preparation.adoc]: Please help confirm proxy usage without problems.

https://curl.haxx.se/docs/manpage.html
https://www.suse.com/support/kb/doc/?id=7006845
https://github.com/SUSE/connect/blob/master/SUSEConnect.8.ronn#use-with-registration-proxy

Comment (jenting) [reply adoc/deployment-preparation.adoc]: Not work...

Seems like right now SUSEConnect supports local SUSE registration proxy (RMT/SMT) only.

Comment (jenting) [reply adoc/deployment-preparation.adoc]: SUSEConnect version `0.3.16`, but cannot find related tag on https://github.com/SUSE/connect

Comment (jenting) [reply adoc/deployment-preparation.adoc]: Verify passed.

- server information with environment variable `http_proxy` with format `http_proxy=http://PROXY_IP_FQDN:PROXY_PORT`
- username/password in file `/root/.curlrc` with format `--proxy-user "username:password"`

**Comment by innobead** [comment]
@nkoranova please help review and merge, thanks!

**Pull request merged**